frontend/src/pages/Admin/Contacts/index.tsx

```diff
@@ -51,27 +182,60 @@ export const Contacts: FC = () => {
       <Renderer
         isLoading={isLoading}
         component={
-          <>
-            <Table
-              columns={data?.columns}
-              rows={data?.rows}
-              loading={isLoading}
-              onRowClick={handleCurrentRowData}
-            />
-          </>
+          <Table
+            columns={data?.columns}
+            rows={data?.rows}
+            loading={isLoading}
+            onRowClick={handleCurrentRowData}
+          />
         }
       />
       <GDXModal
         open={open}
         handleClose={handleClose}
-        modalTitle={
-          "new" === formType ? `New Contact` : `Change Contact ${contactsQuery?.data?.version}`
-        }
+        modalTitle={"new" === formType ? `New Contact` : `Change Contact ${contactQuery?.data?.id}`}
         handleEditMode={handleEditMode}
         editMode={editMode}
         handleFormType={handleFormType}
       >
-        <div>test</div>
+        <>
+          {!editMode ? (
+            <ReadForm fields={readFields} />
+          ) : (
+            <>
+              {"new" === formType ? (
+                <CreateForm
+                  initialValues={createFormInitialValues}
+                  // todo: Define a good type. "Any" type temporarily permitted.
+                  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+                  onSubmit={async (values: any) => {
+                    return handlePost({
+                      formValues: values,
+                      apiUrl: "/contacts",
+                      handleEditMode: handleEditMode,
+                      queryKeys: ["/contacts"],
+                    });
+                  }}
+                  editFields={editFields}
+                />
+              ) : (
+                <EditForm
+                  initialValues={contactQuery?.data}
+                  onSubmit={async (values) => {
+                    return handleUpdate({
+                      changedValues: values,
+                      currentRowData: contactQuery?.data,
+                      apiUrl: `contacts/${contactQuery?.data?.id}`,
+                      handleEditMode: handleEditMode,
+                      queryKeys: [`contact - ${currentRowData?.id}`, `contacts`],
```

Comment:
not sure contacts needs to be there
```suggestion
                      queryKeys: [`contact - ${currentRowData?.id}`],
```